backend/test_observer/controllers/test_executions/test_executions.py

```diff
@@ -190,4 +196,14 @@ def get_test_results(id: int, db: Session = Depends(get_db)):
     if test_execution is None:
         raise HTTPException(status_code=404, detail="TestExecution not found")
 
+    historic_test_executions = get_historic_test_executions(db, test_execution)
+    historic_test_result_mapping = get_historic_test_result_mapping(
+        historic_test_executions
+    )
+
+    for test_result in test_execution.test_results:
+        test_result.historic_results = historic_test_result_mapping[
+            test_result.test_case_id
+        ]
```

Comment:
what would happen if there is a new test case not tested before?

Reply:
Thanks for catching this, it would throw a KeyError. I have now fixed this.